packages/browser/src/index.ts

```diff
@@ -25,4 +44,36 @@ export default class LogtoClient {
   constructor(logtoConfig: LogtoConfig) {
     this.logtoConfig = logtoConfig;
   }
+
+  public async signInWithBrowser(redirectUri: string) {
+    const { clientId, endpoint, requester, resources, scopes: originalScopes } = this.logtoConfig;
+
+    this.oidcConfig = await fetchOidcConfig(
+      `${endpoint}/oidc/.well-known/openid-configuration`,
+      requester
+    );
+
+    const { authorizationEndpoint } = this.oidcConfig;
+    const codeVerifier = generateCodeVerifier();
+    const codeChallenge = await generateCodeChallenge(codeVerifier);
+    const state = generateState();
+    const scopes = withReservedScopes(originalScopes).split(' ');
+
+    const signInUri = generateSignInUri({
+      authorizationEndpoint,
+      clientId,
+      redirectUri,
+      codeChallenge,
+      state,
+      scopes,
+      resources,
+    });
+
+    const logtoKey = getLogtoKey(clientId);
+    const logtoSignInSessionItem: LogtoSignInSessionItem = { redirectUri, codeVerifier, state };
```

Comment:
we can also cache oidc config in session storage (let's add an issue and do it later)

Reply:
Added the linear issue LOG-1519

Reply:
Canceled issue since it's reasonable to just cache oidc config only in object property.

See [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1644809320215889) on Slack.